pkg/oci/utils.go

```diff
@@ -140,10 +143,34 @@ func networkConfig(ocispec spec.Spec) (vc.NetworkConfig, error) {
 
 // PodConfig converts an OCI compatible runtime configuration file
 // to a virtcontainers pod configuration structure.
-func PodConfig(runtime RuntimeConfig, bundlePath, cid, console string) (*vc.PodConfig, *spec.Spec, error) {
+func PodConfig(runtime RuntimeConfig, bundlePath, cid, console string, podAnnotations, containerAnnotations map[string]string) (*vc.PodConfig, *spec.Spec, error) {
+	finalPodAnnotations := make(map[string]string)
+	finalContainerAnnotations := make(map[string]string)
+
+	// copy pod annotations
+	for k, v := range podAnnotations {
+		if k == ociConfigPathKey {
+			return nil, nil, fmt.Errorf("pod annotations map already contains %v: %v", ociConfigPathKey, v)
+		}
+		if k == ociBundlePathKey {
+			return nil, nil, fmt.Errorf("pod annotations map already contains %v: %v", ociBundlePathKey, v)
+		}
+
+		finalPodAnnotations[k] = v
+	}
+
+	// copy container annotations
+	for k, v := range containerAnnotations {
+		finalContainerAnnotations[k] = v
+	}
+
 	configPath := filepath.Join(bundlePath, "config.json")
 	ociLog.Debugf("converting %s", configPath)
 
+	// supplement pod annotations
+	finalPodAnnotations[ociConfigPathKey] = configPath
+	finalPodAnnotations[ociBundlePathKey] = bundlePath
```

Comment:
Ok, I _believe_ that's where the confusion is coming from: The current implementation is wrong, both the bundle and config file paths are container specific artifacts and should be carried through the single container annotations map. Not through pod ones.

At the moment, this routine should return a PodConfig with an empty pod annotations map.